File: src/01-easy/retro-typewriter-art/solution.fs
open System
open System.Collections.Generic
[<EntryPoint>]
let main _ =
    let ab = dict ["sp",' ';"bS",'\\';"sQ",'\'';"nl",'\n']
    let line = Console.ReadLine()
    let sb = Text.StringBuilder()
    for tok in line.Split(' ') do
        let l2 = if tok.Length >= 2 then tok.[tok.Length-2..] else ""
        let ch, num =
            if ab.ContainsKey(l2) then ab.[l2], tok.[..tok.Length-3]
            else tok.[tok.Length-1], tok.[..tok.Length-2]
        let n = if String.IsNullOrEmpty(num) then 1 else max 1 (int num)
        sb.Append(String(ch, n)) |> ignore
    printfn "%s" (sb.ToString())
    0

File: src/01-easy/retro-typewriter-art/solution.rs
use std::io;
use std::collections::HashMap;
fn main(){
    let ab:HashMap<&str,char>=[("sp",' '),("bS",'\\'),("sQ",'\''),("nl",'\n')].into();
    let mut line=String::new(); io::stdin().read_line(&mut line).unwrap();
    let mut out=String::new();
    for tok in line.trim().split(' '){
        let (ch,num)=if tok.len()>=2{
            let l2=&tok[tok.len()-2..];
            if let Some(&c)=ab.get(l2){(c,&tok[..tok.len()-2])}
            else{(tok.chars().last().unwrap(),&tok[..tok.len()-1])}
        }else{(tok.chars().last().unwrap(),"")};
        let n:usize=if num.is_empty(){1}else{num.parse().unwrap_or(1)};
        for _ in 0..n{out.push(ch);}
    }
    println!("{}",out);
}
